packages/contracts/docs/osx/02-how-to-guides/02-plugin-development/04-upgradeable-plugin/04-subsequent-builds.md

```diff
@@ -0,0 +1,141 @@
+
+As you can see, we're still inheritting from the `PluginUUPSUpgradeable` contract and simply overriding some implementation from the previous build. The idea is that when someone upgrades the plugin and calls on these functions, they'll use this new upgraded implementation, rather than the older one.
+
+```solidity
+import {PluginUUPSUpgradeable, IDAO} '@aragon/osx/core/plugin/PluginUUPSUpgradeable.sol';
+
+/// @title SimpleStorage build 2
+contract SimpleStorageBuild2 is PluginUUPSUpgradeable {
+  bytes32 public constant STORE_PERMISSION_ID = keccak256('STORE_PERMISSION');
+
+  uint256 public number; // added in build 1
+  address public account; // added in build 2
+
+  /// @notice Initializes the plugin when build 2 is installed.
+  function initializeBuild2(
+    IDAO _dao,
+    uint256 _number,
+    address _account
+  ) external reinitializer(2) {
+    __PluginUUPSUpgradeable_init(_dao);
+    number = _number;
+    account = _account;
+  }
+
+  /// @notice Initializes the plugin when the update from build 1 to build 2 is applied.
+  /// @dev The initialization of `SimpleStorageBuild1` has already happened.
+  function initializeFromBuild1(address _account) external reinitializer(2) {
+    account = _account;
+  }
+
+  function storeNumber(uint256 _number) external auth(STORE_PERMISSION_ID) {
+    number = _number;
+  }
+
+  function storeAccount(address _account) external auth(STORE_PERMISSION_ID) {
+    account = _account;
+  }
+}
+```
+
+Builds that you publish don't necessarily need to introduce new storage varaibles of your contracts and don't necessarily need to change the storage. To read more about Upgradeability, check out [OpenZeppelin's UUPSUpgradeability implementation here](https://docs.openzeppelin.com/contracts/4.x/api/proxy#UUPSUpgradeable).
+
+### 3. Alternatively, a build implementation modifying bytecode
+
+Updates for your contracts don't necessarily need to affect the storage, they can also modify the plugin's bytecode. Modifying the contract's bytecode, means making changes to:
+
+- functions
+- constants
+- immutables
+- events
+- errors
+
+For this third build, then, we want to change the bytecode of our implementation as an example, so we 've introduced two separate permissions for the `storeNumber` and `storeAccount` functions and named them `STORE_NUMBER_PERMISSION_ID` and `STORE_ACCOUNT_PERMISSION_ID` permission, respectively. Additionally, we decided to add the `NumberStored` and `AccountStored` events as well as an error preventing users from setting the same value twice. All these changes only affect the contract bytecode and not the storage.
+
+Here, it is important to remember how Solidity stores `constant`s (and `immutable`s). In contrast to normal variables, they are directly written into the bytecode on contract creation so that we don't need to worry that the second `bytes32` constant that we added shifts down the storage so that the value in `uint256 public number` gets lost.
```

Comment:
One doubt, are immutables not being stored? It should be a read-only variable, but a variable nonetheless?